src/main/kotlin/org/codefreak/codefreak/util/FileUtil.kt

```diff
@@ -0,0 +1,52 @@
+package org.codefreak.codefreak.util
+
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.attribute.PosixFilePermission
+
+object FileUtil {
+
+  /**
+   * Remove leading dots and slashes from given path and normalizes patterns like `foo/../bar`.
+   */
+  fun sanitizeName(name: String, basePath: String = "/") = Paths.get(basePath, name).normalize().toString().trim().trim('/')
+
+  fun getFilePermissionsMode(permissions: Set<PosixFilePermission>): Int {
+    return permissions.sumOf(FileUtil::getFilePermissionsMode)
+  }
+
+  fun getFilePermissionsMode(permission: PosixFilePermission): Int {
+    return when (permission) {
+      PosixFilePermission.OWNER_READ -> 100
```

Comment:
File permissions are octal numbers. So `100`<sub>8</sub> is actually `64`<sub>10</sub>.